muikku-frontend/muikkuV3/actions/announcements/helpers.ts

```diff
@@ -105,30 +110,37 @@ export async function loadAnnouncementsHelper(
     params.workspaceEntityId = workspaceId;
   }
 
-  switch (item.id) {
-    case "expired":
-      params.timeFrame = "EXPIRED";
-      break;
-    case "unread":
-      params.timeFrame = "ALL";
-      params.onlyUnread = true;
-      break;
-    case "archived":
-      params.timeFrame = "ALL";
-      params.onlyArchived = true;
-      break;
-    case "own":
-      params.timeFrame = "ALL";
-      params.onlyMine = true;
-      break;
-    default:
-      params.timeFrame = "CURRENTANDUPCOMING";
-      break;
+  if (item) {
+    switch (item.id) {
+      case "expired":
+        params.timeFrame = "EXPIRED";
+        break;
+      case "unread":
+        params.timeFrame = "ALL";
+        params.onlyUnread = true;
+        break;
+      case "archived":
+        params.timeFrame = "ALL";
+        params.onlyArchived = true;
+        break;
+      case "own":
+        params.timeFrame = "ALL";
+        params.onlyMine = true;
+        break;
+      default:
+        params.timeFrame = "CURRENTANDUPCOMING";
+        break;
+    }
+  }
+
+  if (category) {
+    params.timeFrame = "ALL";
+    params.categoryIds = [category.id];
   }
 
   try {
     const newAnnouncements = await announcerApi.getAnnouncements(params);
-
+    const categories = await announcerApi.listAnnouncementCategories();
```

Comment:
Is there reason why categories are loaded everytime when loadAnnouncementsHelper is called? Doesn't one initial load be enough.